File: Java/Program/Pemilik.java
//Class Pemilik turunan dari Orang untuk simpan data tambahan atribut
public class Pemilik extends Orang{
    private String kodePemilik;
    private String kodeKosan;
    private String noRekening;

    //Constructor kosong
    public Pemilik(){
        super(); // panggil constructor Orang
        this.kodePemilik = "";
        this.kodeKosan = "";
        this.noRekening = "";
    }

    //Constructor lengkap
    public Pemilik(long noKtp, String nama, String alamat, String noTelp, String kodePemilik, String kodeKosan, String noRekening){
        super(noKtp, nama, alamat, noTelp); // panggil constructor Orang dengan parameter
        this.kodePemilik = kodePemilik;
        this.kodeKosan = kodeKosan;
        this.noRekening = noRekening;
    }

    //Getter & Setter khusus atribut Pemilik
    public String getKodePemilik(){
        return kodePemilik;
    }
    public void setKodePemilik(String kodePemilik){
        this.kodePemilik = kodePemilik;
    }

    public String getKodeKosan(){
        return kodeKosan;
    }
    public void setKodeKosan(String kodeKosan){
        this.kodeKosan = kodeKosan;
    }

    public String getNoRekening(){
        return noRekening;
    }
    public void setNoRekening(String noRekening){
        this.noRekening = noRekening;
    }
}

File: CPP/Program/Pemilik.cpp
#pragma once
#include "Orang.cpp"
#include <string>
using namespace std;

//Class Pemilik turunan dari orang untuk data pemilik kos
class Pemilik : public Orang{

    private:
    string kodePemilik, kodeKosan, noRekening;

    public:
    //Constructor kosong
    Pemilik() : Orang(){
        this->kodePemilik = "";
        this->kodeKosan = "";
        this->noRekening = "";
    }

    //Constructor dengan parameter untuk buat pemilik dengan data
    Pemilik(long long noKtp, string nama, string alamat, string noTelp, string kodePemilik, string kodeKosan, string noRekening)
        //Memanggil constructor Orang
        : Orang(noKtp, nama, alamat, noTelp){
            this->kodePemilik = kodePemilik;
            this->kodeKosan = kodeKosan;
            this->noRekening = noRekening;
        }

    //Getter & Setter untuk atribut khusus Pemilik
    string getKodePemilik(){
        return kodePemilik;
    }
    void setKodePemilik(string kodePemilik){
        this->kodePemilik = kodePemilik;
    }

    string getKodeKosan(){
        return kodeKosan;
    }
    void setKodeKosan(string kodeKosan){
        this->kodeKosan = kodeKosan;
    }

    string getNoRekening(){
        return noRekening;
    }
    void setNoRekening(string noRekening){
        this->noRekening = noRekening;
    }

    //Destruktor
    ~Pemilik(){}
};